experimental/InvariantTheory/src/InvariantTheory.jl

```diff
@@ -675,3 +721,82 @@ function affine_algebra(R::RedGroupInvarRing)
   end
   return domain(R.presentation), R.presentation
 end
+
+
+###################
+#In the case there is no reynolds operator
+###################
+
+#Algorithm 4.5.1 in Computational Invariant Theory by Derksen and Kemper  
+
+function inv_gens_no_reynolds(I::MPolyIdeal, mat_::MatElem, ringg::MPolyRing) 
+    #I is the null cone ideal, mat_ is the representation matrix and 
+    #ringg is the ring in which we must return the invariants
+    g = gens(I)
+    V = Vector{Int}()
+    n = ncols(mat_)
+    #make a vector of all the degrees of generators of the null cone ideal I
+    for gen in g
+        d = total_degree(gen)
+        if !(d in V)
+            push!(V, Int(d))
+        end
+    end
```

Comment:
Should `V` be sorted? 

Note that `d in V` can be expensive if `g` is larger (but I guess this is mostly irrelevant here...). Anyway, this *may* be more efficient:
```suggestion
    n = ncols(mat_)
    #make a vector of all the degrees of generators of the null cone ideal I
    V = map(total_degree,g)
    sort!(V)
    unique!(V)
```